collect/collect.go

```diff
@@ -426,7 +424,14 @@ func (i *InMemCollector) processSpan(sp *types.Span) {
 	// if the trace we got back from the cache has already been sent, deal with the
 	// span.
 	if trace.Sent {
-		i.dealWithSentTrace(cache.NewKeptTraceCacheEntry(trace), sp)
+		if sr, reason, found := i.sampleTraceCache.Check(sp); found {
+			// bump the count of records on this trace -- if the root span isn't
+			// the last late span, then it won't be perfect, but it will be better than
+			// having none at all
+			i.dealWithSentTrace(sr, reason, sp)
+			return
+		}
+		// TODO: would it be possible that it's not in the cache?
```

Comment:
If trace.Sent is true, that means that we found it in the active trace cache and then to get here it would have to not be in the sampleTraceCache. I think in that weird edge case, we should just use "late" as the reason and do something like this:
`i.dealWithSentTrace(cache.NewKeptTraceCacheEntry(trace), "late", sp)`